### PR TITLE
Restore messaging tests

### PR DESCRIPTION
**Pull request by Patrik-Stas**
Signed-off-by: Patrik Stas <patrik.stas@absa.africa>

### CODE REVIEW

File: libvcx/src/messages/get_message.rs [hunk 1]
```diff
@@ -539,55 +540,56 @@ mod tests {
...
-        let credential_offer = ::issuer_credential::issuer_credential_create(cred_def_handle,
-                                                                             "1".to_string(),
-                                                                             institution_did.clone(),
-                                                                             "credential_name".to_string(),
-                                                                             credential_data.to_owned(),
-                                                                             1).unwrap();
+        let (alice_to_faber, faber_to_alice) = ::connection::tests::create_connected_connections();
 
-        ::issuer_credential::send_credential_offer(credential_offer, alice).unwrap();
+        info!("test_download_messages :: going send message from faber to alice");
```
Comment (mirgee) [inline libvcx/src/messages/get_message.rs:550]: "sending messages from Faber to Alice"

File: libvcx/src/messages/get_message.rs [hunk 2]
```diff
@@ -539,55 +540,56 @@ mod tests {
...
+        send_generic_message(faber_to_alice, "Hello Alice", &json!({"msg_type": "toalice", "msg_title": "msg1"}).to_string()).unwrap();
+        info!("test_download_messages :: messge sent");
```
Comment (mirgee) [inline libvcx/src/messages/get_message.rs:552]: messages sent

File: libvcx/src/messages/get_message.rs [hunk 3]
```diff
@@ -539,55 +540,56 @@ mod tests {
...
+        assert!(all_messages[0].msgs[1].decrypted_payload.is_some());
+
+        let received = download_messages(None, Some(vec![MessageStatusCode::Received.to_string()]), None).unwrap();
+        assert_eq!(received.len(), 1);
+        assert_eq!(received[0].msgs.len(), 2);
+        assert!(received[0].msgs[0].decrypted_payload.is_some());
+        assert_eq!(received[0].msgs[0].status_code, MessageStatusCode::Received);
+        assert!(received[0].msgs[1].decrypted_payload.is_some());
+
+        // there should be review aries message connections/1.0/response from Aries-Faber connection protocol
```
Comment (mirgee) [inline libvcx/src/messages/get_message.rs:576]: Mark this as TODO

File: libvcx/src/messages/get_message.rs [hunk 4]
```diff
@@ -539,55 +540,56 @@ mod tests {
...
+        assert_eq!(specific[0].msgs.len(), 1);
 
-        // No pending will return empty list
-        let empty = download_messages(None, Some(vec!["MS-103".to_string()]), Some(vec![accepted[0].msgs[0].uid.clone()])).unwrap();
-        assert_eq!(empty.len(), 1);
-
-        let hello_msg = download_messages(None, None, Some(vec![hello_uid])).unwrap();
-        assert_eq!(hello_msg[0].msgs[0].decrypted_payload, Some("{\"@type\":{\"name\":\"hello\",\"ver\":\"1.0\",\"fmt\":\"json\"},\"@msg\":\"hello\"}".to_string()));
-
-        // Agency returns a bad request response for invalid dids
```
Comment (mirgee) [inline libvcx/src/messages/get_message.rs:588]: Does this require changes on agency side?

**Review by mirgee — CHANGES_REQUESTED** [review]

File: libvcx/src/messages/get_message.rs (hunk 2 above)
Comment (Patrik-Stas) [reply libvcx/src/messages/get_message.rs]: I've deleted logs from the test

File: libvcx/src/messages/get_message.rs (hunk 4 above)
Comment (Patrik-Stas) [reply libvcx/src/messages/get_message.rs:588]: Node agency is currently not performing did validation, it just tries to match the did. We should probably audit agency inputs and probably think about its error reporting in general...

File: libvcx/src/messages/get_message.rs (hunk 3 above)
Comment (Patrik-Stas) [reply libvcx/src/messages/get_message.rs]: this is misunderstanding, I've clarified the comment

**Review by mirgee — APPROVED** [review]

**Review by mirgee — APPROVED** [review]

**Pull request merged**